pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -870,24 +870,36 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 				unmountDeviceErr)
 		}
 
-		inUseErr := util.CheckDeviceInUseBySystem(deviceToDetach.DevicePath)
-		if inUseErr != nil {
-			if os.IsNotExist(inUseErr) {
+		// Before logging that UnmountDevice succeeded and moving on,
+		// use util.DeviceOpened to check if the device is in use anywhere
+		// else on the system. Retry if it returns true.
+		deviceOpened, deviceOpenedErr := util.DeviceOpened(deviceToDetach.DevicePath)
+		if deviceOpenedErr != nil {
+			if os.IsNotExist(deviceOpenedErr) {
 				// If device path does not exist, print warning but do not fail out.
+				// Currently only hit in tests where fakes do not have device paths.
+				// In other cases, still want to gracefully exit unmount and continue.
 				glog.Infof(
-					"WARNING: CheckDeviceInUseBySystem failed for volume %q (spec.Name: %q) with: %v",
+					"WARNING: UnmountDevice.DeviceOpened failed for volume %q (spec.Name: %q) with: %v",
 					deviceToDetach.VolumeName,
 					deviceToDetach.VolumeSpec.Name(),
-					inUseErr)
+					deviceOpenedErr)
 			} else {
-				// On failure, return error. Caller will log and retry.
+				// There was an error in the call to DeviceOpened. Caller will log and retry.
 				return fmt.Errorf(
-					"CheckDeviceInUseBySystem failed for volume %q (spec.Name: %q) with: %v",
+					"UnmountDevice.DeviceOpened failed for volume %q (spec.Name: %q) with: %v",
 					deviceToDetach.VolumeName,
 					deviceToDetach.VolumeSpec.Name(),
-					inUseErr)
+					deviceOpenedErr)
 			}
 		}
+		// The device is still in use elsewhere. Caller will log and retry.
+		if deviceOpened {
+			return fmt.Errorf(
+				"UnmountDevice failed for volume %q (spec.Name: %q) because the device is in use",
```

Comment:
Please reword log line to be extra clear. 

"because the device is in use when we it was no longer expected to be in use."